libhoney.go

```diff
@@ -120,6 +120,30 @@ type Config struct {
 	Transport http.RoundTripper
 }
 
+// VerifyWriteKey calls out to the Honeycomb API to validate the write key, so
+// we can exit immediately if desired instead of happily sending events that
+// are all rejected.
+func VerifyWriteKey() error {
+	url := fmt.Sprintf("%s/1/team_slug", defaultBuilder.APIHost)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("User-Agent", UserAgentAddition)
+	req.Header.Add("X-Honeycomb-Team", defaultBuilder.WriteKey)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		return errors.New("Non-200 response verifying Honeycomb write key")
```

Comment:
As a user, this would be a frustratingly vague error to get. I'd suggest distinguishing a few cases:
- empty write key (can check before making HTTP request)
- valid write key (HTTP status 200)
- invalid write key (HTTP status 401)
- network or other error (any other HTTP status, or error making the request)